rpc/src/main/scala/org/bitcoins/rpc/client/RpcClient.scala

```diff
@@ -0,0 +1,1114 @@
+      amounts: Map[BitcoinAddress, Bitcoins],
+      minconf: Int = 1,
+      comment: String = "",
+      subtractFeeFrom: Vector[BitcoinAddress] = Vector.empty): Future[
+    DoubleSha256Digest] = {
+    bitcoindCall[DoubleSha256Digest]("sendmany",
+                                     List(JsString(""),
+                                          Json.toJson(amounts),
+                                          JsNumber(minconf),
+                                          JsString(comment),
+                                          Json.toJson(subtractFeeFrom)))
+  }
+
+  def sendToAddress(
+      address: BitcoinAddress,
+      amount: Bitcoins,
+      localComment: String = "",
+      toComment: String = "",
+      subractFeeFromAmount: Boolean = false): Future[DoubleSha256Digest] = {
+    bitcoindCall[DoubleSha256Digest](
+      "sendtoaddress",
+      List(JsString(address.toString),
+           JsNumber(amount.toBigDecimal),
+           JsString(localComment),
+           JsString(toComment),
+           JsBoolean(subractFeeFromAmount))
+    )
+  }
+
+  def sendRawTransaction(
+      transaction: Transaction,
+      allowHighFees: Boolean = false): Future[DoubleSha256Digest] = {
+    bitcoindCall[DoubleSha256Digest](
+      "sendrawtransaction",
+      List(JsString(transaction.hex), JsBoolean(allowHighFees)))
+  }
+
+  def setAccount(address: BitcoinAddress, account: String): Future[Unit] = {
+    bitcoindCall[Unit]("setaccount",
+                       List(JsString(address.value), JsString(account)))
+  }
+
+  def setBan(
+      address: URI,
+      command: String,
+      banTime: Int = 86400,
+      absolute: Boolean = false): Future[Unit] = {
+    bitcoindCall[Unit]("setban",
+                       List(JsString(address.getAuthority),
+                            JsString(command),
+                            JsNumber(banTime),
+                            JsBoolean(absolute)))
+  }
+
+  def setNetworkActive(activate: Boolean): Future[Unit] = {
+    bitcoindCall[Unit]("setnetworkactive", List(JsBoolean(activate)))
+  }
+
+  // Should be BitcoinFeeUnit
+  def setTxFee(feePerKB: Bitcoins): Future[Boolean] = {
+    bitcoindCall[Boolean]("settxfee", List(JsNumber(feePerKB.toBigDecimal)))
+  }
+
+  def signMessage(address: P2PKHAddress, message: String): Future[String] = {
+    bitcoindCall[String]("signmessage",
+                         List(JsString(address.value), JsString(message)))
+  }
+
+  def signMessageWithPrivKey(
+      key: ECPrivateKey,
+      message: String): Future[String] = {
+    bitcoindCall[String]("signmessagewithprivkey",
+                         List(JsString(key.toWIF(network)), JsString(message)))
+  }
+
+  private def signRawTransaction(
+      transaction: Transaction,
+      utxoDeps: Option[Vector[RpcOpts.SignRawTransactionOutputParameter]],
+      keys: Option[Vector[ECPrivateKey]],
+      sigHash: Option[String]): Future[SignRawTransactionResult] = {
+    val params =
+      if (utxoDeps.isEmpty) {
```

Comment:
Let's avoid nesting here. Break this out into top level statements. 

``` 
val utxos = utxos.map(u => Json.toJson(u)
.getOrElse(Vector.empty)
val keys = keys.map(k => Json.toJson(k))
.getOrElse(Vectory.empty)
val sighash = sigHash.map(s => Json.toJson(s))
.getOrElse(Vector.empty)
```
unless you can't define `keys` without a `utxo`